ApplicationUnitTests/Commands/Authentication/RegisterHandlerTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Identity_Application.Commands.Authentication;
+using Identity_Application.Interfaces.Authentication;
+using Identity_Application.Models.AuthorizationModels;
+using Moq;
+
+namespace ApplicationUnitTests.Commands.Authentication;
+
+public class RegisterHandlerTests
+{
+    private readonly Mock<IAuthenticationService> _mockAuthenticationService;
+    private readonly RegisterHandler _handler;
+
+    public RegisterHandlerTests()
+    {
+        _mockAuthenticationService = new Mock<IAuthenticationService>();
+        _handler = new RegisterHandler(_mockAuthenticationService.Object);
+    }
+
+    [Fact]
+    public async Task Handle_ValidCommand_ShouldRegisterUser()
```

Comment:
how about scenario when invalid input? 
```
var errors = _validator.Validate(request.RegisterVM);

        foreach ( var error in errors.Errors )
        {
            throw new Exception(error.ErrorMessage);
        }
```

Reply:
If model is totally wrong - it is going to be null, we have check for that. For other incorrect inputs we have the validation.